src/main/java/gregtech/api/capability/impl/SteamMultiWorkable.java

```diff
@@ -59,126 +55,42 @@ protected void trySearchNewRecipe() {
+                exportInventory,
+                exportFluids,
+                MAX_PROCESSES,
+                maxVoltage);
+        this.applyBuilderFeatures(builder);
+        if (builder != null) {
+            return builder.build().getResult();
+        } else {
             return null;
         }
-
-        return recipeMap.recipeBuilder()
-                .inputsIngredients(recipeInputs)
-                .outputs(recipeOutputs)
-                .EUt(Math.min(32, (int) Math.ceil(recipeEUt * 1.33)))
-                .duration(Math.max(recipeDuration, (int) (recipeDuration * (100.0F / (100.0F + speedBonusPercent)) * 1.5)))
-                .build().getResult();
     }
 
-    private void computeOutputItemStacks(Collection<ItemStack> recipeOutputs, ItemStack outputStack, int recipeAmount) {
-        if (!outputStack.isEmpty()) {
-            int finalAmount = outputStack.getCount() * recipeAmount;
-            int maxCount = outputStack.getMaxStackSize();
-            int numStacks = finalAmount / maxCount;
-            int remainder = finalAmount % maxCount;
-
-            for (int fullStacks = numStacks; fullStacks > 0; fullStacks--) {
-                ItemStack full = outputStack.copy();
-                full.setCount(maxCount);
-                recipeOutputs.add(full);
-            }
-
-            if (remainder > 0) {
-                ItemStack partial = outputStack.copy();
-                partial.setCount(remainder);
-                recipeOutputs.add(partial);
+    @Override
+    public void applyBuilderFeatures(RecipeBuilder<?> builder) {
+        if (builder == null) {
+            this.invalidInputsForRecipes = true;
+        } else {
+            if (builder.getParallel() == 0) {
+                this.isOutputsFull = true;
+            } else {
+                this.parallelRecipesPerformed = builder.getParallel();
+                //apply coil bonus
+                applyParallelBonus(builder);
             }
         }
     }
 
-    // This does no checking to see if outputs can fit, similar to other chanced output only recipes
-    private Recipe doChancedOnlyRecipe(Recipe matchingRecipe, ItemStack stack) {
-        RecipeBuilder<?> builder = recipeMap.recipeBuilder()
-                .inputs(new CountableIngredient(matchingRecipe.getInputs().get(0).getIngredient(), stack.getCount()))
-                .EUt(Math.min(32, (int) Math.ceil(matchingRecipe.getEUt() * 1.33)))
-                .duration((int) (matchingRecipe.getDuration() * 1.5));
-
-        Recipe.ChanceEntry entry = matchingRecipe.getChancedOutputs().get(0);
-        int maxProcesses = Math.min(MAX_PROCESSES, stack.getCount());
-        for (int i = 0; i < maxProcesses; i++)
-            builder.chancedOutput(entry.getItemStack(), entry.getChance(), entry.getBoostPerTier());
-
-        return builder.build().getResult();
+    @Override
+    public void applyParallelBonus(RecipeBuilder<?> builder) {
+        builder.EUt(5).duration(192);
```

Comment:
I think the names of those are really bad but cant think of anything better.
applyBuilderFeatures was meant to basically move all the notifiable logic from the main method.
applyParallelBonus is this case applies the energy/duration bonus/penalties. But say in case of the processing array it would do nothing.
As i wrote this probably this doesnt work too well with the distinct logic.